src/client.rs

```diff
@@ -113,14 +118,22 @@ impl Client {
     }
 
     // Request must be signed
-    fn sign_request(&self, endpoint: &str, request: &str) -> String {
-        let mut signed_key = Hmac::<Sha256>::new_varkey(self.secret_key.as_bytes()).unwrap();
-        signed_key.update(request.as_bytes());
-        let signature = hex_encode(signed_key.finalize().into_bytes());
-        let request_body: String = format!("{}&signature={}", request, signature);
-        let url: String = format!("{}{}?{}", self.host, endpoint, request_body);
-
-        url
+    fn sign_request(&self, endpoint: API, request: Option<String>) -> String {
+        match request {
+            Some(request) => {
+                let mut signed_key = Hmac::<Sha256>::new_varkey(self.secret_key.as_bytes()).unwrap();
+                signed_key.update(request.as_bytes());
+                let signature = hex_encode(signed_key.finalize().into_bytes());
+                let request_body: String = format!("{:?}&signature={}", request, signature);
+                format!("{}{}?{}", self.host, String::from(endpoint), request_body)
+            },
+            None => {
+                let signed_key = Hmac::<Sha256>::new_varkey(self.secret_key.as_bytes()).unwrap();
+                let signature = hex_encode(signed_key.finalize().into_bytes());
+                let request_body: String = format!("&signature={}", signature);
+                format!("{}{}?{}", self.host, String::from(endpoint), request_body)
```

Comment:
This line is common in both arms of the match, you can pull it out

Reply:
They are different, on this one `signed_key.update(request.as_bytes())` is not called and the format from the `request_body` is different.

Reply:
I was only referring to the last line which is common; the rest is indeed different. But it's a minor point.